dashboard/app/models/user_level.rb

```diff
@@ -186,6 +186,21 @@ def self.update_lockable_state(user_id, level_id, script_id, locked, readonly_an
     user_level.save!(touch: false)
   end
 
+  def self.update_best_result(user_id, level_id, script_id, best_result, touch_updated_at = true)
+    user_level = UserLevel.find_by(
+      level_id: level_id,
+      script_id: script_id,
+      user_id: user_id
+    )
+
+    if user_level.present?
+      user_level.best_result = best_result
+
+      # touch_updated_at=false preserves updated_at, which represents the user's submission timestamp.
```

Comment:
Do we need to adjust / remove this comment?

Reply:
I was trying to explain what this value is for, it is a confusing comment though. If you think the intent is clear enough from the code, then I'll delete the comment

Reply:
Ah, ok, let's just leave it then.